ide/extexecution.base/src/org/netbeans/api/extexecution/base/BaseExecutionService.java

```diff
@@ -140,6 +140,43 @@ public static BaseExecutionService newService(@NonNull Callable<? extends Proces
         return new BaseExecutionService(processCreator, descriptor);
     }
 
+    /**
+     * Infers the output encoding from the relevant system properties, if those should all be <code>null</code>
+     * then this will fallback to <code>Charset.defaultCharset()</code>
+     * 
+     * Since JDK 18 and JEP400 Console.charset() is used for the console's encoding instead of <code>Charset.defaultCharset()</code>. 
+     * Console.charset() is exposed via stdout.encoding/sun.stdout.encoding.
+     * If ran with JDK<=16 these properties should be null and the old default of <code>Charset.defaultCharset()</code> will be used to match pre JEP400 behavior.
+     * 
+     * The checking order for the encoding is stdout.encoding, sun.stdout.encoding, native.encoding, <code>Charset.defaultCharset()</code>
+     * 
+     * @return inferred encoding as Charset
+     */
+    private static Charset getInputOutputEncoding(){
+        String[] encodingSystemProperties = new String[]{"stdout.encoding, sun.stdout.encoding, native.encoding"};
+
+        Charset preferredCharset = null;
+        for(String encodingProperty : encodingSystemProperties){
+            String encodingPropertyValue = System.getProperty(encodingProperty);
+            if(encodingPropertyValue == null){
+                continue;
+            }
+
+            try {
+                preferredCharset = Charset.forName(encodingPropertyValue);
```

Comment:
Isn't this missing a `break`?!

Could also chain the variant of System::getProperty with default values to get rid of the loop - pros and cons to validating each value.

Reply:
Where is the location where a `break` is missing?

If the `encodingPropertyValue` is `null` it will cycle through the next property, if the property is filled it will be returned on line 171 if set, or am I missing something?